src/main/actions.js

```diff
@@ -62,3 +63,39 @@ export function setupActions(store) {
     store.subscribe(debounce(() => setupMenu(store), 500));
   });
 }
+
+function handleAuthCall(args) {
+  log.info(args);
+  if (!Array.isArray(args) || args.length === 0) {
+    log.warn('Empty auth call. Abborting.');
+    return;
+  }
+  switch (args[0]) {
+    case 'google':
+      BrowserWindow.getAllWindows().forEach(win => {
+        win.webContents.send('protocol:auth/google', args.slice(1));
+      });
+      break;
+    default:
+      log.warn('Unable to handle google authentication result.');
+      break;
+  }
+}
+
+export function handleProtocolCall(url) {
+  if (typeof url !== 'string' || url.length === 0) {
+    log.warn('Empty protocol call. Abborting.');
+    return;
+  }
+  const path = url.replace('buttercup://', '');
+  const parts = path.split('/');
```

Comment:
Could this be destructured instead, like say `const [ action ] = path.split('/');` ?

Reply:
@perry-mitchell I could do this: `const [action, ...args] = ` because I need the args a few lines below